src/Symfony/Component/Cache/Adapter/FilesystemAdapter.php

```diff
@@ -147,7 +147,7 @@ protected function doSave(array $values, $lifetime)
     private function getFile($id, $mkdir = false)
     {
         $hash = str_replace('/', '-', base64_encode(md5($id, true)));
```

Comment:
So for the proba part, worse case on windows: the 128bits MD5 being turned to case insensitive base64 means `38^21*4 = 6e+33` key space cardinality. On NTFS that handles at most `2^32-1` files per partition, this means that filling at 100% has `1/1.4e+24` chance of generating a collision. I'd say this is enough space for any use cases here.